app/controllers/foreman_ansible/concerns/hosts_controller_extensions.rb

```diff
@@ -23,8 +23,13 @@ def action_permission
 
       def play_roles
         find_resource
-        task = async_task(::Actions::ForemanAnsible::PlayHostRoles, @host)
-        redirect_to task
+        @composer = JobInvocationComposer.for_feature(:ansible_run_host, @host)
+        if @composer.save
+          @composer.trigger
+          redirect_to job_invocation_path(@composer.job_invocation)
+        else
+          raise ::Foreman::Exception.new(N_("Could not run Ansible roles for %s") % @host)
```

Comment:
IMHO we shouldn't raise an exception here, most likely there are validation problems which is why composer was not saved, since we rescue from this exception on line below, it seems we're using exceptions for controlling the flow. Compose should not raise Foreman::Exceptions, so I think we should just add the redirection to else branch. If there's other exception, it's probably code error and should case 500.